src/components/Modals/SendManaModal/SendManaModal.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect, useState } from 'react'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import { Button, Close, Field } from 'decentraland-ui'
+import Modal from 'decentraland-dapps/dist/containers/Modal'
+import { Props } from './SendManaModal.types'
+import './SendManaModal.css'
+
+const TransactionDetailModal: React.FC<Props> = ({
+  name,
+  onClose,
+  isLoading,
+  manaPrice,
+  onManaPrice,
+  onSendMana
+}) => {
+  const [amount, setAmount] = useState(0)
+  const [to, setTo] = useState('')
+  const [errors, setErrors] = useState({
+    amount: { hasError: false, message: '' },
+    to: { hasError: false, message: '' }
+  })
+
+  const handleSetAmount = (e: React.FormEvent<HTMLInputElement>) => {
+    const intValue = parseInt(e.currentTarget.value, 10)
+    if (e.currentTarget.value.length === 0) {
+      setAmount(0)
+    } else if (!isNaN(intValue)) {
+      setAmount(intValue)
+    }
+  }
+
+  const handleSetTo = (e: React.FormEvent<HTMLInputElement>) => {
+    const { value } = e.currentTarget
+    const isValid = /^[0-9a-fA-Fx]{0,42}$/.test(value)
+    if (isValid) {
+      setErrors({
+        ...errors,
+        to: { hasError: false, message: '' }
+      })
+      setTo(value)
+    } else {
+      setErrors({
+        ...errors,
+        to: {
+          hasError: true,
+          message: t('send_mana_modal.errors.invalid_char')
+        }
+      })
+    }
+  }
+
+  const handleSendMana = () => {
+    const isValidAddress = /^0x[0-9a-fA-Fx]{40}$/.test(to)
+    if (isValidAddress) {
+      onSendMana(to, amount)
+    } else {
+      setErrors({
+        ...errors,
+        to: {
+          hasError: true,
+          message: t('send_mana_modal.errors.invalid_address')
+        }
+      })
+    }
+  }
+
+  useEffect(() => {
```

Comment:
this will fetch on every render, since it doesn't have specified dependencies. If we just want to fetch the mana price when the component is mounted it should be:

```ts
useEffect(() => {
  onManaPrice()
}, [])
```

Notice the empty array as second argument, which means that effect will only be triggered once.